DFReader.py

```diff
@@ -555,6 +555,105 @@ def set_message_timestamp(self, m):
         m._timestamp = self.timebase + count/rate
 
 
+class DFMetaData(object):
+    '''handle dataflash messages metadata'''
+    def __init__(self, parent):
+        self.parent = parent
+        self.data = None
+        self.metadata_load_attempted = False
+
+    def reset(self):
+        '''clear cached data'''
+        self.data = None
+        self.metadata_load_attempted = False
+
+    def dot_pymavlink(self, *args):
+        '''return a path to store mavproxy data'''
+        if 'HOME' not in os.environ:
+            dir = os.path.join(os.environ['LOCALAPPDATA'], '.pymavlink')
+        else:
+            dir = os.path.join(os.environ['HOME'], '.pymavlink')
+        if len(args) == 0:
+            return dir
+        return os.path.join(dir, *args)
```

Comment:
@peterbarker - It feels like this should be in mavutil or something for consistent use by all pymavlink functions.
(for example one place that accesses .mavproxy specifies "~" - which looks very Linux specific!)
Would you agree, and if so, shall I do so on this PR, or on a pre-PR, or a later PR?